tests/bluetooth/audio/bap_broadcast_source/src/main.c

```diff
@@ -1283,3 +1345,110 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base)
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_base(fixture->source, NULL);
+	zassert_not_equal(0, err, "Did not fail with null ID");
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
+}
+
+ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base_inval_state)
+{
+	struct bt_bap_broadcast_source_param *create_param = fixture->param;
+	struct bt_bap_broadcast_source *source;
+	int err;
+
+	NET_BUF_SIMPLE_DEFINE(base_buf, 64);
+
+	printk("Creating broadcast source with %zu subgroups with %zu streams\n",
+	       create_param->params_count, fixture->stream_cnt);
+
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	source = fixture->source;
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
+
+	err = bt_bap_broadcast_source_get_base(source, &base_buf);
+	zassert_not_equal(0, err, "Did not fail with deleted broadcast source");
+}
+
+ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base_inval_very_small_buf)
+{
+	struct bt_bap_broadcast_source_param *create_param = fixture->param;
+	int err;
+
+	NET_BUF_SIMPLE_DEFINE(base_buf, 15); /* Too small to hold any BASE */
+
+	printk("Creating broadcast source with %zu subgroups with %zu streams\n",
+	       create_param->params_count, fixture->stream_cnt);
+
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_base(fixture->source, &base_buf);
+	zassert_not_equal(0, err, "Did not fail with too small base_buf (%u)", base_buf.size);
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
+}
+
+ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_get_base_inval_small_buf)
+{
+	struct bt_bap_broadcast_source_param *create_param = fixture->param;
+	int err;
+
+	/* Can hold a base, but not large enough for this configuration */
+	NET_BUF_SIMPLE_DEFINE(base_buf, 64);
+
+	printk("Creating broadcast source with %zu subgroups with %zu streams\n",
+	       create_param->params_count, fixture->stream_cnt);
+
+	err = bt_bap_broadcast_source_create(create_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
+
+	err = bt_bap_broadcast_source_get_base(fixture->source, &base_buf);
+	zassert_not_equal(0, err, "Did not fail with too small base_buf (%u)", base_buf.size);
+
+	err = bt_bap_broadcast_source_delete(fixture->source);
+	zassert_equal(0, err, "Unable to delete broadcast source: err %d", err);
+	fixture->source = NULL;
+}
```

Comment:
It looks like those two are testing exactly the same. Are you sure we need separate tests for `very_small_buf` and `small_buf`?
I do not get which buffer size values can be considered as very small, and which are small ^^

Reply:
Right, so the "very small buf" is a buffer that is smaller than `MINIMUM_BASE_SIZE` and fails as part of 
```c
	if ((buf->size - buf->len) < MINIMUM_BASE_SIZE) {
		return false;
	}
```
in `encode_base`.

The "small buf" is a test that verifies that the buffer supplied can hold a minimum BASE, but not the BASE we supply, and will fail in one of the dynamic checks like
```c
	if ((buf->size - buf->len) < sizeof(len)) {
		LOG_DBG("No room for metadata length");

		return false;
	}
```

So the two tests different error conditions (and thus more code coverage), and therefore I believe both should exist.